### PR TITLE
Corner cases handling

### PR DESCRIPTION
**Pull request by piotrmilcarz**

### CODE REVIEW

**Review by artursmirnov — CHANGES_REQUESTED** [review]
Hey @piotrmilcarz, thank you for the contribution! I have some comments code-wise, please take a look when you have a chance.

Apart from that, could you please also add a simple acceptance test that briefly checks that warning message shows up when it should and that modal window works as expected.

Also, could you please provide some instructions that would allow to test this PR in action? We have pricing branch deployed to staging API and billing. Although, it seems like there's some more set up needs to be done, does it?

Thank you!

File: app/templates/components/dialogs/switch-to-free-modal.hbs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            as |field|
+          >
+            <field.textarea rows="4" placeholder="Could you share more information with us?">{{this.switchToFreeReasonDetails}}</field.textarea>
+          </form.field>
+        </div>
+        {{log }}
+        <div class="form-elem switchtofree-notice">
+          You are switching to the Free Trier Plan. Your remaining addon credits will be discarded and only the credits coming with the Free Trier Plan plan will be available.
+          {{#if (or (gt this.subscription.addonUsage.public.remainingCredits 0)
+                    (gt this.subscription.addonUsage.private.remainingCredits 0))}}
```
[the remark this answers is not in the document]
Comment (piotrmilcarz) [reply app/templates/components/dialogs/switch-to-free-modal.hbs:37]: 👍

File: app/models/v2-subscription.js [hunk 2]
```diff
@@ -111,6 +111,13 @@ export default Model.extend({
     return yield this.api.post(`/v2_subscription/${this.id}/pay`);
   }).drop(),
 
+  switchToFreeSubscription: task(function* (data) {
+    yield this.api.post(`/v2_subscription/${this.id}/changetofree`, {
+      data
+    });
+    yield this.accounts.fetchV2Subscriptions.perform();
+  }).drop(),
```
[the remark this answers is not in the document]
Comment (artursmirnov) [reply app/models/v2-subscription.js:127]: This feedback is not implemented. @piotrmilcarz it should be easy to implement, could you please take a look?

[the remark this answers is not in the document]
Comment (piotrmilcarz) [reply app/models/v2-subscription.js]: I had to miss it.
Refactored now.

**Review by artursmirnov — APPROVED** [review]
@piotrmilcarz looks good to me now! Thank you for implementing the feedback 👍

File: app/templates/components/dialogs/switch-to-free-modal.hbs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        <div class="form-elem switchtofree-information">
+          <form.field
+            @label=""
+            @onChange={{action (mut this.switchToFreeReasonDetails)}}
+            as |field|
+          >
+            <field.textarea rows="4" placeholder="Could you share more information with us?">{{this.switchToFreeReasonDetails}}</field.textarea>
+          </form.field>
+        </div>
+        {{log }}
```
[the remark this answers is not in the document]
Comment (artursmirnov) [reply app/templates/components/dialogs/switch-to-free-modal.hbs:33]: Could you please remove this one? 
```suggestion
```

**Review by artursmirnov — APPROVED** [review]

**Pull request merged**